analysis/README.md

```diff
@@ -0,0 +1,623 @@
++               pol constant p_X_const = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_X_read_free = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_Y_const = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_Y_read_free = [0, 0, 1, 0, 0] + [0]*;
++               pol constant p_instr__jump_to_operation = [0, 1, 0, 0, 0] + [0]*;
++               pol constant p_instr__loop = [0, 0, 0, 0, 1] + [1]*;
++               pol constant p_instr__reset = [1, 0, 0, 0, 0] + [0]*;
++               pol constant p_instr_identity = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_instr_nothing = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_instr_one = [0, 0, 1, 0, 0] + [0]*;
++               pol constant p_instr_return = [0, 0, 0, 1, 0] + [0]*;
++               pol constant p_read_X_A = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_read_X_pc = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_read_Y_A = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_read_Y_pc = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_reg_write_X_A = [0, 0, 0, 0, 0] + [0]*;
++               pol constant p_reg_write_Y_A = [0, 0, 1, 0, 0] + [0]*;
++               { pc, reg_write_X_A, reg_write_Y_A, instr_identity, instr_one, instr_nothing, instr__jump_to_operation, instr__reset, instr__loop, instr_return, X_const, X_read_free, read_X_A, read_X_pc, Y_const, Y_read_free, read_Y_A, read_Y_pc } in { p_line, p_reg_write_X_A, p_reg_write_Y_A, p_instr_identity, p_instr_one, p_instr_nothing, p_instr__jump_to_operation, p_instr__reset, p_instr__loop, p_instr_return, p_X_const, p_X_read_free, p_read_X_A, p_read_X_pc, p_Y_const, p_Y_read_free, p_read_Y_A, p_read_Y_pc };
++       }
++
+-       function main {
+-               start::
+-               A <=Y= one();
+-               // END BATCH Unimplemented
+-               return;
+-               // END BATCH
++       function main<2> {
++
+```
+
+### Block enforcer
+
+This step takes constrained machines and enforces that the `function_id` can only change if the `latch` is on. This defines blocks of computation which can be created based on the functions exposed by each machine.
+
+We add an identical block of constraints for each machine type:
+```diff
++       constraints {
++               pol constant _block_enforcer_last_step = [0]* + [1];
++               pol commit _function_id_no_change;
++               _function_id_no_change = ((1 - _block_enforcer_last_step) * (1 - instr_return));
++               (_function_id_no_change * (_function_id' - _function_id)) = 0;
++       }
++
++       }
++
++       constraints {
++               pol constant _block_enforcer_last_step = [0]* + [1];
++               pol commit _function_id_no_change;
++               _function_id_no_change = ((1 - _block_enforcer_last_step) * (1 - instr_return));
++               (_function_id_no_change * (_function_id' - _function_id)) = 0;
+```
+
```

Comment:
This looks a bit off. The `}` on line 265 is extra and the `{` online 269 isn't closed. Also these are identical, as explained above. But how does the compiler treat this?

Reply:
I adjusted the display for the diff to be clearer. I don't understand the last question?

Reply:
I think I get it now. Each of these identical blocks are part of different machines, so there is no conflict. Maybe it makes sense to just show one block here (instead of 2) and just say each machine gets one of them (which is already said).